xbmc/epg/GUIEPGGridContainer.cpp

```diff
@@ -1429,9 +1429,11 @@ bool CGUIEPGGridContainer::OnMouseWheel(char wheel, const CPoint &point)
 
 CPVRChannelPtr CGUIEPGGridContainer::GetChannel(int iIndex)
 {
+  CSingleLock lock(m_critSection);
+
   if (iIndex >= 0 && (size_t) iIndex < m_channelItems.size())
   {
-    CFileItemPtr fileItem = m_channelItems[iIndex];
+    const CFileItemPtr &fileItem = m_channelItems[iIndex];
```

Comment:
Please explain how this breaks ref counting. I don't see that.
The commit message explains why this is an improvement. Now you're saying this isn't one. Do you disagree with my commit message?